From 2c51e8e03c79d3dc36abd01e89ce0a621b0170cd Mon Sep 17 00:00:00 2001
From: Nicolas Bigaouette <nbigaouette@gmail.com>
Date: Thu, 25 Nov 2010 17:15:56 -0500
Subject: [PATCH] Fix valgrind error:
 ==13695== Conditional jump or move depends on uninitialised value(s)
 ==13695==    at 0x4CE608: NcVar::NcVar(NcFile*, int) (in /home/nicolas/fichiers/md.git/mdgit)
 ==13695==    by 0x4D1981: NcFile::NcFile(char const*, NcFile::FileMode, unsigned long*, unsigned long, NcFile::FileFormat) (in /home/nicolas/fichiers/md.git/mdgit)
 ==13695==
 ==13695== Conditional jump or move depends on uninitialised value(s)
 ==13695==    at 0x4C27D0F: strcpy (mc_replace_strmem.c:311)
 ==13695==    by 0x4CE63F: NcVar::NcVar(NcFile*, int) (in /home/nicolas/fichiers/md.git/mdgit)
 ==13695==    by 0x4D1981: NcFile::NcFile(char const*, NcFile::FileMode, unsigned long*, unsigned long, NcFile::FileFormat) (in /home/nicolas/fichiers/md.git/mdgit)

---
 cxx/netcdf.cpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/cxx/netcdf.cpp b/cxx/netcdf.cpp
index 6cb0e74..68d341e 100644
--- a/cxx/netcdf.cpp
+++ b/cxx/netcdf.cpp
@@ -1448,6 +1448,7 @@ NcVar::NcVar(NcFile* nc, int id)
    : NcTypedComponent(nc), the_id(id)
 {
     char nam[NC_MAX_NAME];
+    memset(nam, 0, NC_MAX_NAME*sizeof(char));
     if (the_file 
 	&& NcError::set_err(
 			    nc_inq_varname(the_file->id(), the_id, nam)
-- 
1.7.3.2

